### DISCUSSION TITLE
[Question] how do I create a BStr from &[u8] ?

### DISCUSSION THREAD

**Discussion by jtmoon79** [BurntSushi/bstr#107] [Q&A] [answered]
Newbie question: how do I create a `bstr::BStr` instance from a `&[u8]`?

Given `let a: [u8; 3] = [ 40, 46, 55 ];`, I've tried variations
- `bstr::BStr::From::<&[u8]>(&a)`
- `bstr::BStr::From(&a)`
- `bstr::BStr::From(a)`
- `bstr::BStr::new(&a)`
- `bstr::BStr::from_bytes(&a)`

I reviewed the docs and it wasn't obvious to this rust newbie.

**Comment by lopopolo** [comment, marked as answer]
I think this is how you're meant to do it:

```rust
use bstr::ByteSlice;

let a = [0_u8; 3];
let b = a.as_bstr();
```

Most of the APIs in this crate come from either `ByteSlice` or `ByteVec`. These are extension traits, which means when they are imported, they make methods available on vecs and slices that `bstr` implements these traits on.

**Comment by BurntSushi** [comment]
It's also worth pointing out that you don't actually have a byte slice here. You have an array of bytes. It's likely we could improve the from impls with const generics.

**Comment by BurntSushi** [comment]
For const generics, see #86. Otherwise moving this to a Discussion Q&A.
